web/src/app/chat/lib.tsx

```diff
@@ -21,6 +21,22 @@ import { Persona } from "../admin/assistants/interfaces";
 import { ReadonlyURLSearchParams } from "next/navigation";
 import { SEARCH_PARAM_NAMES } from "./searchParams";
 
+export async function updateChatSession(
```

Comment:
nit: prefer more descriptive name e.g. `updateModelOverrideForChatSession`

Reply:
This also avoids the overloading of the name, since the hook also has something with this same name